pkg/csi/service/vanilla/controller.go

```diff
@@ -596,77 +597,148 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 		ContentSourceSnapshotID: contentSourceSnapshotID,
 	}
 
+	// Check if vCenter task for this volume is already registered as part of
+	// improved idempotency CR
+	log.Debugf("Checking if vCenter task for volume %s is already registered.", req.Name)
 	var (
-		sharedDatastores []*cnsvsphere.DatastoreInfo
+		volTaskAlreadyRegistered bool
+		volumeInfo               *cnsvolume.CnsVolumeInfo
+		faultType                string
+		vcenter                  *cnsvsphere.VirtualCenter
 	)
-	// Get accessibility.
-	topologyRequirement := req.GetAccessibilityRequirements()
-	if topologyRequirement != nil {
-		// Check if topology domains have been provided in the vSphere CSI config secret.
-		// NOTE: We do not support kubernetes.io/hostname as a topology label.
-		if c.manager.CnsConfig.Labels.TopologyCategories == "" && c.manager.CnsConfig.Labels.Zone == "" &&
-			c.manager.CnsConfig.Labels.Region == "" {
-			return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
-				"topology category names not specified in the vsphere config secret")
-		}
+	// Get VirtualCenter instance
+	vcenter, err = c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
+	if err != nil {
+		return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+			"failed to get vCenter. Err: %v", err)
+	}
+	// Get operation store
+	operationStore := c.manager.VolumeManager.GetOperationStore()
+
+	volumeOperationDetails, err := operationStore.GetRequestDetails(ctx, req.Name)
+	if err == nil && volumeOperationDetails.OperationDetails != nil {
```

Comment:
Log if there was a error

Reply:
Added a debug message. Kept level as debug, as we will always get this message for the first time since task is not yet registered.